packages/react/src/components/header/components/headerNavigationMenu/HeaderNavigationMenu.tsx

```diff
@@ -24,40 +23,59 @@ export type HeaderNavigationMenuProps = React.PropsWithChildren<{
   id?: string;
 }>;
 
+const activeLinkClassName = classNames(
+  styles.headerNavigationMenuLinkContent,
+  styles.headerNavigationMenuLinkContentActive,
+);
+
+const renderHeaderNavigationMenuItem = (child, index) => {
+  const linkContainerClasses = child.props.active ? activeLinkClassName : styles.headerNavigationMenuLinkContent;
+  const className = classNames(child.props.className, styles.headerNavigationMenuLink);
+  const node = cloneElement(child as React.ReactElement, {
+    className,
+    index,
+  });
+
+  return (
+    // eslint-disable-next-line react/no-array-index-key
+    <li key={index} className={styles.headerNavigationMenuLinkContainer}>
+      <span className={linkContainerClasses}>{node}</span>
+    </li>
+  );
+};
+
+const HeaderNavigationMenuContent = () => {
+  const { navigationContent } = useHeaderContext();
+  return (
+    <>
+      {Children.map(navigationContent, (child, index) => {
+        if (!isValidElement(child)) return null;
+        return renderHeaderNavigationMenuItem(child, index);
```

Comment:
Something is giving the mobile nav menu a double top border. Maybe the first list item?